adapters/operaads/operaads.go

```diff
@@ -0,0 +1,232 @@
+package operaads
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/prebid/prebid-server/macros"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+var (
+	ErrEmptyPlacementId = errors.New("empty placement id")
+	ErrEmptyEndpointId  = errors.New("empty endpoint id")
+	ErrEmptyPublisherId = errors.New("empty publisher id")
+)
+
+type OperaadsAdapter struct {
```

Comment:
We prefer the name `adapter`. There is no need to repeat the package name. There is no need to export this type by starting with an uppercase letter.